Libraries/LibCrypto/Authentication/HMAC.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2025, Altomani Gianluca <altomanigianluca@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCrypto/Authentication/HMAC.h>
+
+#include <openssl/core_names.h>
+#include <openssl/evp.h>
+
+namespace Crypto::Authentication {
+
+HMAC::HMAC(Hash::HashKind hash_kind, ReadonlyBytes key)
+    : m_hash_kind(hash_kind)
+    , m_key(key)
+    , m_mac(EVP_MAC_fetch(nullptr, "HMAC", nullptr))
+{
+    reset();
+}
+
+HMAC::~HMAC()
+{
+    EVP_MAC_free(m_mac);
+    EVP_MAC_CTX_free(m_ctx);
+}
+
+size_t HMAC::digest_size() const
+{
+    return EVP_MAC_CTX_get_mac_size(m_ctx);
+}
+
+void HMAC::update(u8 const* message, size_t length)
+{
+    if (EVP_MAC_update(m_ctx, message, length) != 1) {
+        VERIFY_NOT_REACHED();
```

Comment:
isn't this slightly too much, i.e, wouldn't it be better to bail out with an error and continue the operation instead? 🤔

Reply:
There is no way to bail out of this function currently. The hashes implementation does the same, I guess we'll need to do some `ErrorOr`.